src/ftdc_tools/rollups/client_perf.py

```diff
@@ -82,7 +110,9 @@ def add_doc(self, doc: FTDCDoc) -> None:
         self._gauges_workers_max = min(
             self._gauges_workers_max, doc["gauges"]["workers"]
         )
-        self._extracted_durations.append(extracted_duration)
+        self._extracted_durations = self._extracted_durations + number_of_ops * [
+            extracted_duration / number_of_ops
+        ]
```

Comment:
While I like the elegance here, I think this list concatenation might be less efficient (both memory and CPU-wise) than generating the new list and repeatedly appending to `self._extracted_durations`. It'd be similar to when we were using dataframes: a new one being allocated and repeatedly iterating over it which killed performance